kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -110,10 +116,13 @@ def application(self) -> KInner:
     fields: FrozenDict
 
     def __init__(self, contract_name: str, contract_json: Dict, foundry: bool = False) -> None:
-        def _get_method_abi(_mname: str) -> Dict:
+        def _get_method_abi(_mname: str, _margs: List[str]) -> Dict:
             for _method in contract_json['abi']:
                 if _method['type'] == 'function' and _method['name'] == _mname:
-                    return _method
+                    if len(_margs) == len(_method['inputs']):
+                        inputs_match = [(_method['inputs'][i]['type'] == _margs[i]) for i in range(len(_margs))]
+                        if all(inputs_match):
+                            return _method
```

Comment:
```suggestion
                    _marg_types = (_marg['type'] for _marg in _method['inputs'])
                    if len(_margs) == len(_marg_types) and all(a1 == a2 for a1, a2 in zip(_margs, _marg_types, strict=True)):
                        return _method
```

Reply:
Thanks, this looks like a better way to write it. I incorporated this in the latest commit.